test/unit/view_model/record_test.rb

```diff
@@ -285,31 +323,47 @@ def validate!
 
     describe 'with read-only attribute' do
       let(:attributes) { { read_only: { read_only: true } } }
+      let(:default_model_values) { { read_only: 'immutable' } }
+      let(:default_view_values) { { read_only: 'attempted change' } }
 
-      include CanSerialize
-      include CanDeserializeToExisting
+      describe 'asserting the default' do
+        let(:default_view_values) { default_model_values }
 
-      it 'deserializes to new without the attribute' do
-        new_view = default_view.tap { |v| v.delete('read_only') }
-        vm = viewmodel_class.deserialize_from_view(new_view, deserialize_context: create_context)
-        refute(default_model.equal?(vm.model))
-        assert_nil(vm.model.read_only)
-        assert_edited(vm, new: true)
-      end
+        include CanSerialize
+        include CanDeserializeToExisting
 
-      it 'rejects deserialize from new' do
-        ex = assert_raises(ViewModel::DeserializationError::ReadOnlyAttribute) do
-          viewmodel_class.deserialize_from_view(default_view, deserialize_context: create_context)
+        it 'deserializes to new with the attribute' do
+          vm = viewmodel_class.deserialize_from_view(default_view, deserialize_context: create_context)
+          assert_equal(default_model, vm.model)
+          refute(default_model.equal?(vm.model))
+          assert_edited(vm, new: true)
+        end
+
+        it 'deserializes to new without the attribute' do
+          new_view = default_view.tap { |v| v.delete('read_only') }
+          vm = viewmodel_class.deserialize_from_view(new_view, deserialize_context: create_context)
+          assert_equal(default_model, vm.model)
+          refute(default_model.equal?(vm.model))
+          assert_edited(vm, new: true)
         end
-        assert_equal('read_only', ex.attribute)
       end
 
-      it 'rejects update if changed' do
-        new_view = default_view.merge('read_only' => 'written')
-        ex = assert_raises(ViewModel::DeserializationError::ReadOnlyAttribute) do
-          viewmodel_class.deserialize_from_view(new_view, deserialize_context: update_context)
+      describe 'attempting a change' do
+        let(:default_view_values) { { read_only: 'attempted change' } }
```

Comment:
It's been quite a while since I read these tests, but I think expressing it like this is making it harder to reason about.

IIUC, the`default_view` is a "neutral" serialization that is to be used as a base for the tests to extend with their exceptional cases. To make an attempted change, as also seen in the previous version of the test, you add your change to the neutral view: `new_view = default_view.merge(read_only: 'attempted_change')`.

In this case where the value is used twice, I'd prefer a `let(:new_view) { ... }`.

Reply:
I agree, and have changed in the new refactor